From 444b479df37421c2a24c9cd74ab944a371c12c6f Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@kde.org>
Date: Mon, 24 Aug 2026 16:21:29 +0200
Subject: [PATCH] QIconLoader: Don't consider fallbackThemeName() in themeName()

themeName and fallbackThemeName are independent things.

bebbf84167a4341eed27ecfca480ff62dfa1b159 mixes the two together,
which results in setFallbackThemeName() incorrectly changing
themeName().

Restore that part to how it was before.

Amends bebbf84167a4341eed27ecfca480ff62dfa1b159

Fixes: QTBUG-149431
Change-Id: I7bb5e4329b677abb95b74269877f8d9c0222a972
Reviewed-by: Axel Spoerl <axel.spoerl@qt.io>
(cherry picked from commit 85420d5d4f78a6ba4cf70ed8ee092e0ef4fb9a54)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit e80e3f0cebae9c3a45a1b7ce81d6454c699d89c6)
---

diff --git a/src/gui/image/qiconloader.cpp b/src/gui/image/qiconloader.cpp
index 62353ec..4526c6f 100644
--- a/src/gui/image/qiconloader.cpp
+++ b/src/gui/image/qiconloader.cpp
@@ -151,8 +151,7 @@
         return m_userTheme;
 
     if (m_systemTheme.isEmpty()) {
-        const QString &themeName = systemThemeName();
-        m_systemTheme = !themeName.isEmpty() ? themeName : fallbackThemeName();
+        m_systemTheme = systemThemeName();
     }
     return m_systemTheme;
 }
